llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -649,6 +754,12 @@ bool SpecConstantsPass::collectSpecConstantMetadata(Module &M,
 
     IDMap[ID] = Descs;
   }
+  unsigned Index = 0, Offset = 0;
+  for (const auto *Node : N->operands()) {
+    auto *Constant = cast<ConstantAsMetadata>(Node->getOperand(0))->getValue();
+    collectCompositeElementsDefaultValuesRecursive(M, Constant, Index, Offset,
+                                                   DefaultValues);
+  }
```

Comment:
Default values here won't be used in native mode, so probably it makes sense to split this method into two and call the second one conditionally to avoid inserting an empty property set into device images